src/libsyntax/codemap.rs

```diff
@@ -592,16 +592,21 @@ impl CodeMap {
         CodeMap {
             files: RefCell::new(Vec::new()),
             expansions: RefCell::new(Vec::new()),
-            file_loader: file_loader
+            file_loader: RefCell::new(file_loader)
         }
     }
 
+    pub fn set_file_loader(&self, file_loader: Box<FileLoader>) {
+        let mut loader = self.file_loader.borrow_mut();
+        *loader = file_loader;
+    }
+
```

Comment:
Using a different entry point, but I see what you've done. I think I can remove this change, although I need to add some other changes to my code to do the same thing.